app/src/main/java/org/schabi/newpipe/util/URLHandler.java

```diff
@@ -0,0 +1,121 @@
+package org.schabi.newpipe.util;
+
+import android.content.Context;
+
+import org.schabi.newpipe.extractor.NewPipe;
+import org.schabi.newpipe.extractor.StreamingService;
+import org.schabi.newpipe.extractor.exceptions.ExtractionException;
+import org.schabi.newpipe.extractor.exceptions.ParsingException;
+import org.schabi.newpipe.extractor.linkhandler.LinkHandlerFactory;
+import org.schabi.newpipe.extractor.stream.StreamInfo;
+import org.schabi.newpipe.player.playqueue.PlayQueue;
+import org.schabi.newpipe.player.playqueue.SinglePlayQueue;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
+import io.reactivex.rxjava3.core.Single;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+public final class URLHandler {
+
+    private URLHandler() {
+    }
+
+    /**
+     * Check if an URL can be handled in NewPipe.
+     * <p>
+     * This method will check if the provided url can be handled in NewPipe or not. If this is a
+     * service URL with a timestamp, the popup player will be opened.
+     * <p>
+     * The timestamp param accepts two integers, corresponding to two timestamps types:
+     * 0 for {@code &t=} (used for timestamps in descriptions),
+     * 1 for {@code #timestamp=} (used for timestamps in comments).
+     * Any other value of this integer will return false.
+     *
+     * @param context       the context to be used
+     * @param url           the URL to check if it can be handled
+     * @param timestampType the type of timestamp
+     * @return true if the URL can be handled by NewPipe, false if it cannot
+     */
+    public static boolean canHandleUrl(final Context context,
```

Comment:
I was wrong with renaming this method.
It actually returns whether the URL was handled by NewPipe (either by opening a popup or through the RouterActivity). So the method needs to be re-renamed and the JDoc altered, too. Sorry for the inconvinience. 

-------

**Note for the future:** We should not play the video in a popup by default. Instead, if astream which represents the given url is currently playing, the current timestamp should be modified. 
Otherwise we could do one of the following
1. open the stream in popup,
1. enqueue the stream and start playing it,
1. ask the user what to do with the stream (e.g. when long-pressing the link) 

But if I didn't miss anything, this needs some deep refactoring to make the player queues available in NavigationHelper or the PlayerHelper.

Reply:
> But if I didn't miss anything, this needs some deep refactoring to make the player queues available in NavigationHelper or the PlayerHelper.

Yes, always open popup player when clicking on any timestamp is a bad idea. However, I will not take care of changing this.